In [2]:
import numpy as np
import pandas as pd
import scipy.stats as stats
from scipy.stats import ttest_1samp,ttest_ind, mannwhitneyu,levene,shapiro,wilcoxon
from statsmodels.stats.power import ttest_power
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import statistics as st

In [3]:
golf=pd.read_csv('SM4.csv')

In [33]:
golf.describe()

Current         New
count   40.000000   40.000000
mean   270.275000  267.500000
std      8.752985    9.896904
min    255.000000  250.000000
25%    263.000000  262.000000
50%    270.000000  265.000000
75%    275.250000  274.500000
max    289.000000  289.000000

In [42]:
golf.isna().sum()

Current    0
New        0
dtype: int64

In [6]:
Current=golf['Current']
New=golf['New']

****Calculating the Population Parameters****


In [8]:
pop_mean_Current = Current.mean()
pop_std_Current = st.stdev(Current)
print("Population Mean: "+str(pop_mean_Current))
print("Population Standard Deviation: "+str(pop_std_Current))

Population Mean: 270.275
Population Standard Deviation: 8.752984838884595


****Calculating the Sample Parameters****

In [10]:
print("Population Mean: "+str(New.mean()))
golf_New_sample = New.sample(frac=0.10)
sample_mean_New = golf_New_sample.mean()
print("Sample Mean: "+str(sample_mean_New))
sample_std_New = st.stdev(golf_New_sample)
print("Sample Standard Deviation: "+str(sample_std_New))

Population Mean: 267.5
Sample Mean: 262.5
Sample Standard Deviation: 4.509249752822894


****Hypothesis:****

**H0**: Distance covered Current ball and New ball doesnt change 

**Ha**: Distance covered by Current and New ball are not equal

****Population Parameters:****

Mean=pop_mean_Current

Std=pop_std_Current

****Sample Parameters:****

Mean=sample_mean_New

Std=sample_std_New

In [14]:
Z_Stat=((sample_mean_New - pop_mean_Current )* np.sqrt(4))/sample_std_New

In [15]:
Z_Stat

-3.448467229003072

Conclusion: As the test statistics fall into the rejection region the null hypothesis is rejected
and it can be concluded that the distance of new ball decreases.
However, we will first check for the Type 1 and Type 2 Errors

In [58]:
pop_mean_New = New.mean()
pop_std_New = st.stdev(New)

zscore_error = ((pop_mean_New - pop_mean_Current)*np.sqrt(40)/pop_std_New)
print(zscore_error)

-1.7733465125020238


In [71]:
stats.norm.cdf(-1.77)

0.03836357036287122

Since it is two tailed test at Confidence interval of 95% level of significance becomes .025 each side for which z critical is -1.95 and 1.95 , hence we see that z_stat is < z critical hence we can not reject null hypothesis, hence we have committed Type 1 error

P-Val: .038

****At Current Confidence Interval Mean Lies Between****

In [92]:
print("There is 95% Probability that True Population Mean of Current Ball Distance is in Between"+str(round(((-1.95 * pop_std_New)/np.sqrt(40))+ pop_mean_Current,2)) +str(' & ') +str(round(((1.95 * pop_std_New)/np.sqrt(40))+ pop_mean_Current,2)))


There is 95% Probability that True Population Mean of Current Ball Distance is in Between267.22 & 273.33


In [96]:
print("There is 95% Probability that True Population Mean of New   Ball Distance is in Between" +str(round((((1.95 * pop_std_New)/np.sqrt(40)) - pop_mean_New)*-1,2)) +str(' & ') +str(round((((-1.95 * pop_std_New)/np.sqrt(40)) - pop_mean_New)*-1,2)))

There is 95% Probability that True Population Mean of New   Ball Distance is in Between264.45 & 270.55


There is 95%probability that the true population mean is between 267.22 and 273.33 (New Ball Distance)

There is 95%probability that the true population mean is between 264.45 and 270.55 (Current Ball Distance)

And by their confidence intervals, we can say that there is no significance difference between the driving distances of Current Model and New Model. 

****Non-Parameteric 2 Sample test****

In [55]:
u,P_Val=mannwhitneyu(Current,New)

In [57]:
P_Val

0.10265192122893124

****Power Test****

In [42]:
stdpooled=np.sqrt((((40-1)*Current.var())+((40-1)*New.var()))/(40+40-2))

In [43]:
stdpooled

9.342469200871404

In [44]:
zcal=(New.mean()- Current.mean())/stdpooled

In [45]:
zcal

-0.2970306821820877

In [54]:
ttest_power(-.297,nobs=40,alpha=.05,alternative='two-sided')

0.44919938117044644

There are 44.9% chance for me to reject Ho when it be false

****Levene Test****

In [72]:
levene(Current,New)

LeveneResult(statistic=0.2553238291765741, pvalue=0.6147759523331309)

Since pvalue frome levene test is greater than the level of significance , we say that variance in both samples are equal that is we accept the null hypothesis which states varation in distance measured for current and new ball are similar 

****Shapiro Test****

In [99]:
print("Normality Test For New " +str(shapiro(New)))
print("Normality Test For Current" +str(shapiro(Current)))

Normality Test For New (0.9678265452384949, 0.3064688742160797)
Normality Test For Current(0.9707045555114746, 0.37878310680389404)


Null Hypothesis for shapiro test:

Ho: Population is normally Distributed

Ha: Population is not normally Distributed

Since pvalue for both population Current and New is greater than the level of significance , it states that our Ho is not rejected

****Conclusion****

My recommendation for the company Par.Inc. is to continue with further study of the improved cut-resistant, longer lasting golf ball. But ofcourse, Par, Inc. should also consider other factors like costing,availability of suppliers and the like before finalizing and launching the product. With this Newly developed ball considering it's only given factor i.e. distance we conclude that it wont have any significant effect on market share that is it wont increase Par inc. market share. For concrete conclusion Par inc. should provide other variables apart from distance